src/main/java/teammates/ui/webapi/GetNotificationsAction.java

```diff
@@ -63,10 +63,37 @@ public JsonResult execute() {
                     logic.getActiveNotificationsByTargetUser(targetUser);
         }
 
-        NotificationsData responseData = new NotificationsData(notificationAttributes);
+        boolean isFetchingAll = Boolean.parseBoolean(
+                getRequestParamValue(Const.ParamsNames.NOTIFICATION_IS_FETCHING_ALL));
+        if (!isFetchingAll) {
+            // only unread notifications are returned if user is not fetching all
+            List<String> readNotifications = logic.getReadNotificationsId(userInfo.getId());
+            notificationAttributes = notificationAttributes
+                    .stream()
+                    .filter(n -> !readNotifications.contains(n.getNotificationId()))
+                    .collect(Collectors.toList());
+        }
```

Comment:
Correct me if I'm wrong, if the request is from an admin `isFetchingAll` will default to false and this block of code will run. Although the final outcome will not be affected (it will still return all notifications) as admin does not keep track of read notifications, we could probably early return the response for admin before this block.

Reply:
yes i will add a check for admin for this block as well

Reply:
I think you can early return inside the block following the check for `targetUserString == null && userInfo.isAdmin` (line 49)

Reply:
Okay got it